[TESTING ONLY] Use redispatch mechanism to redispatch for make_dual and fw_primal

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73854
* #73853
* #73852
* __->__ #73705
* #73655

Fixes: https://github.com/pytorch/pytorch/issues/73557

TODO
- Not too sure why this works yet, presumably if we use the official redispatch mechanism instead of just at::blah, autograd/ADInplaceOrView kernels will be reenabled?
- LoggingTensor doesn't work for some reason still (it uses no_dispatch... but that shouldn't matter because we only have a single TensorWrapper anyway)

soulitzer committed:
Use redispatch mechanism to redispatch for make_dual and fw_primal

[ghstack-poisoned]

soulitzer committed:
Update on "Use redispatch mechanism to redispatch for make_dual and fw_primal"




[ghstack-poisoned]

soulitzer committed:
Update on "Use redispatch mechanism to redispatch for make_dual and fw_primal"


Fixes: https://github.com/pytorch/pytorch/issues/73557


[ghstack-poisoned]

albanD commented on torch/csrc/autograd/VariableTypeManual.cpp:
```diff
@@ -333,7 +333,7 @@ namespace ADInplaceOrView {
     auto tmp = ([&]() {
       at::AutoDispatchBelowADInplaceOrView guard;
-      return at::alias(self);
+      return at::_ops::alias::redispatch(ks & c10::after_ADInplaceOrView_keyset, self);
```
We discussed this a while with Brian and this is not ok.
You should never redispstch if you call a different function or change any of the arguments.

soulitzer committed:
Update on "Use redispatch mechanism to redispatch for make_dual and fw_primal"


Fixes: https://github.com/pytorch/pytorch/issues/73557

TODO
- Not too sure why this works yet, presumably if we use the official redispatch mechanism instead of just at::blah, autograd/ADInplaceOrView kernels will be reenabled?
- LoggingTensor doesn't work for some reason still (it uses no_dispatch... but that shouldn't matter because we only have a single TensorWrapper anyway)


[ghstack-poisoned]
